bsp/worker/src/mill/bsp/worker/MillBuildServer.scala

```diff
@@ -103,15 +52,17 @@ class MillBuildServer(
         idToModule match {
           case None =>
             val modules: Seq[Module] =
-              ModuleUtils.transitiveModules(evaluator.rootModule) ++ Seq(`mill-build`)
+              ModuleUtils.transitiveModules(evaluator.rootModule) ++ millBuildRootModules
             val map = modules.collect {
-              case m: MillBuildModule =>
+              case m: MillBuildRootModule =>
                 val uri = sanitizeUri(m.millSourcePath)
                 val id = new BuildTargetIdentifier(uri)
-                log.debug(s"mill-build segments: ${`mill-build`.millModuleSegments.render}")
+                for(millBuild <- millBuildRootModules){
+                  log.debug(s"mill-build segments: ${millBuild.millModuleSegments.render}")
+                }
                 (id, m)
               case m: BspModule =>
-                val uri = sanitizeUri(`mill-build`.millSourcePath / m.millModuleSegments.parts)
+                val uri = sanitizeUri(??? : os.Path)
```

Comment:
@lefou I'm not sure what to do here. Other places it seemed straightforward to turn the singleton `mill-build` into a list of `millBuildRootModules`, even though I don't know what the code is trying to do, but over here it's unclear to me what the code doing so I'm not sure how to proceed

Reply:
The point is to have a unique buildTarget ID as an URI. We currently use module segments  and prefix them with the full path of the project, that's why `mill-build.millSourcePath` is used here. We can replace that with some other way to access the project base directory to get rid of the use of the sigleton `mill-build` here.